src/fathom.js

```diff
@@ -0,0 +1,189 @@
+ * message to the page's content script, only if the page url matches any
+ * listener matchPattern. The client content script will call
+ * window.webScience.fathom.addTrainees to add their rulesets. When the fathom
+ * content script receives an isClassifiable message, it will pull these
+ * rulesets from the global window to run Fathom on the page.
+ *
+ * Once the content script is done with classification, a fathomData message
+ * containing the results is sent to the background script, which contains the
+ * results. These messages are then processed using the client's callbacks.
+ *
+ * @module webScience.fathom
+ */
+
+import {ruleset, rule, dom, type, score, out, utils} from 'fathom-web';
+const {isVisible, linearScale} = utils;
+
+import * as pageManager from "./pageManager.js";
+import * as messaging from "./messaging.js";
+import * as events from "./events.js";
+import * as matching from "./matching.js";
+import * as inline from "./inline.js";
+import fathomContentScript from "./content-scripts/fathom.content.js";
+
+/**
+ * Fathom classification results sent by the content script.
+ * @typedef {Object} FathomDataObject
+ *
+ * TODO: Finalize data schema
+ * Likely would involve the URL, pageID, and the fathom classification results.
+ * Currently, just sends an object containing every element with greater than
+ * 0.5 confidence score (key), along with their feature vector and actual score (value).
+ */
+
+/**
+ * The listener specified by the client.
+ * @callback fathomDataListener
+ * @param {FathomDataObject} Results of the Fathom classification on a page.
+ */
+
+/**
+ * @typedef {Object} FathomDataListenerRecord
+ * @property {matching.MatchPatternSet} matchPatternSet - The match patterns for the listener.
+ * @property {browser.contentScripts.RegisteredContentScript} contentScript - The content
+ * script associated with the listener.
+ */
+
+/**
+ * A map where the key is a listener function and each value is a record for that listener function.
+ * @constant {Map<fathomDataListener, FathomDataListenerRecord>}
+ * @private
+ */
+const fathomDataListeners = new Map();
+
+/**
+ * An event that fires when a page's contents has been successfully classified
+ * through a Fathom content script. 
+ *
+ * @constant onFathomData
+ */
+export const onFathomData = events.createEvent({
+    name: "webScience.fathom.onFathomData",
+    addListenerCallback: addListener,
+    removeListenerCallback: removeListener,
+    notifyListenersCallback: () => { return false; }
+});
+
+/**
+ * Whether the module has completed initialization.
+ * @type{boolean}
+ * @private
+ */
+let initialized = false;
+
+
+/**
+ * A callback function to add a fathomDataListener.
+ * @param {fathomDataCallback} listener - The listener that is being removed.
+ * @param {Object} options - Options for the listener.
+ * @param {string[] options.matchPatterns} matchPatterns - The match patterns 
+ * for pages where the listener should be notified.
+ * @private
+ */
```

Comment:
Firefox recently added dynamic imports for content scripts. That would be a good way to handle rulesets and avoid the awkwardness with content script globals. Something like...
* A parameter to `addListener` specifies a path for one or more ruleset JS files and names for each ruleset.
* The ruleset JS file must be a module that exports a `ruleset()` function.
* When a page visit starts, check the URL against the match pattern set for each listener. Send an array of ruleset names and file paths to the content script.
* In the content script, import the relevant rulesets, run them, and send back the results.